versioned/persist/store/src/main/java/org/projectnessie/versioned/persist/store/PersistVersionStore.java

```diff
@@ -150,22 +151,34 @@ public Hash commit(
       }
     }
 
-    commitAttempt.commitMetaSerialized(storeWorker.getMetadataSerializer().toBytes(metadata));
+    commitAttempt.commitMetaSerialized(serializeMetadata(metadata));
 
     return databaseAdapter.commit(commitAttempt.build());
   }
 
   @Override
   public void transplant(
-      BranchName targetBranch, Optional<Hash> referenceHash, List<Hash> sequenceToTransplant)
+      BranchName targetBranch,
+      Optional<Hash> referenceHash,
+      List<Hash> sequenceToTransplant,
+      Function<METADATA, METADATA> updateCommitMetadata)
       throws ReferenceNotFoundException, ReferenceConflictException {
-    databaseAdapter.transplant(targetBranch, referenceHash, sequenceToTransplant);
+    databaseAdapter.transplant(
+        targetBranch,
+        referenceHash,
+        sequenceToTransplant,
+        updateCommitMetadata(updateCommitMetadata));
   }
 
   @Override
-  public void merge(Hash fromHash, BranchName toBranch, Optional<Hash> expectedHash)
+  public void merge(
+      Hash fromHash,
+      BranchName toBranch,
+      Optional<Hash> expectedHash,
+      Function<METADATA, METADATA> updateCommitMetadata)
       throws ReferenceNotFoundException, ReferenceConflictException {
-    databaseAdapter.merge(fromHash, toBranch, expectedHash);
+    databaseAdapter.merge(
```

Comment:
It's because those commits need to be persisted atomically (all or nothing). And also of course conflict detection, finding the common ancestor and such.